doc/source/flexibility/compressed_columns.rst

```diff
@@ -0,0 +1,213 @@
+As this table or tables are of the data dictionary kind, concurrent reads are
+allowed, but writes are serialized, and reads are blocked by writes. Table read
+through old read views are unsupported, similarly to |InnoDB| internal DDL
+transactions. The enforcement of these constraints depends on the
+implementation option below.
+
+Compression dictionaries can be represented as a list of words in a form of a
+string (comma or any other character can be used as a delimiter although not
+required). In other words, ``a,bb,ccc``, ``a bb ccc`` and ``abbccc`` will have
+the same effect. However, the latter is more space-efficient. Quote symbol
+quoting is handled by regular SQL quoting. Dictionaries of at least 128KB
+length are supported.
+
+INFORMATION_SCHEMA Tables
+=========================
+
+This feature implemented two new ``INFORMATION_SCHEMA`` tables.
+
+.. table:: INFORMATION_SCHEMA.XTRADB_ZIP_DICT
+
+  :column BIGINT(21)_UNSIGNED id: dictionary ID
+  :column VARCHAR(64) name: dictionary name
+  :column BLOB zip_dict: compressed dictionary
+
+This table stores the compression dictionary itself. ``SUPER`` privilege is
+required to query it.
+
+.. table:: INFORMATION_SCHEMA.XTRADB_ZIP_DICT_COLS
+
+  :column BIGINT(21)_UNSIGNED table_id: table ID
+  :column BIGINT(21)_UNSIGNED column_pos: column position
+  :column BIGINT(21)_UNSIGNED dict_id: dictionary ID
+
+This table stores the mapping between the compression dictionary and columns
+using the dictionary.
+
+Limitations
+===========
+
+Compressed columns cannot be used in indices (neither on their own nor as parts
+of composite keys).
+
+.. note::
+
+  ``CREATE TABLE t2 AS SELECT * FROM t1``, where ``t1`` already has compressed
+  columns (with or without dictionaries), does not create compressed columns in
+  ``t2``. They can be turned on later by an ``ALTER TABLE`` statement. The
+  behavior is exactly the same as for other column formats. For instance,
+
+  ``CREATE TABLE t1(a BLOB COLUMN_FORMAT DYNAMIC) ENGINE=InnoDB;``
+  ``CREATE TABLE t2 AS SELECT * FROM t1;``
+  will not generate a DYNAMIC column a in ``t2``
+
+  At the same time, after executing ``CREATE TABLE t2 LIKE t1`` statement,
+  ``t2.a`` will have ``COMPRESSED`` attribute.
+
+mysqldump command line parameters
+=================================
+
+By default, with no additional options, ``mysqldump`` will generate compatible
+SQL output.
+All ``/*!50633 COLUMN_FORMAT COMPRESSED */`` and ``/*!50633 COLUMN_FORMAT
+COMPRESSED WITH COMPRESSION_DICTIONARY <dictionary> */`` will be stripped out.
+
+When a new option :option:`enable-compressed-columns` is specified, all
+``/*!50633 COLUMN_FORMAT COMPRESSED */`` will be left intact and all ``/*!50633
+COLUMN_FORMAT COMPRESSED WITH COMPRESSION_DICTIONARY <dictionary> */`` will be
+transformed into ``/*!50633 COLUMN_FORMAT COMPRESSED */``. In this mode no
+``CREATE/DROP COMPRESSION_DICTIONARY ...`` statements will ever be generated.
```

Comment:
Likewise: the dump will contain the necessary SQL statements to create compressed columns, but without dictionaries